samples/net/sockets/echo_client/src/echo-client.c

```diff
@@ -182,3 +172,71 @@ void main(void)
 		stop_tcp();
 	}
 }
+
+static void do_service(k_work_handler_t handler)
+{
+	static struct k_work work;
+
+	k_work_init(&work, handler);
+
+	k_work_submit(&work);
+}
+
+static void event_handler(struct net_mgmt_event_callback *cb,
+			  u32_t mgmt_event, struct net_if *iface)
+{
+	if (mgmt_event == NET_EVENT_L4_CONNECTED) {
+		LOG_INF("Network connected");
+
+		do_service(start_udp_and_tcp);
```

Comment:
If we get multiple NET_EVENT_L4_CONNECT, does the application still work since start_udp_and_tcp will be called more than once?

Reply:
Application cannot receive multiple connect events for a given network interface. There are checks in `net_conn_mgr_connect()` found in `subsys/net/ip/connection_mgr.c` that prevent that.

Reply:
Ok, thanks for the explanation.